casper/src/main/scala/coop/rchain/casper/util/DagOperations.scala

```diff
@@ -91,37 +91,80 @@ object DagOperations {
     })
   }
 
+  private val blockMetadataByNumDecreasing: Ordering[BlockMetadata] =
+    (l: BlockMetadata, r: BlockMetadata) => {
+      def compareByteString(l: ByteString, r: ByteString): Int =
+        l.hashCode().compareTo(r.hashCode())
+
+      val ln = l.blockNum
+      val rn = r.blockNum
+      rn.compare(ln) match {
```

Comment:
I've spent 15 minutes looking for why this thing sorts in descending order. A comment saying 'notice the inverted order of compared items, which makes the ordering descending' would hopefully help me.